src/k2/dto/SKVRecord.h

```diff
@@ -59,16 +65,42 @@ struct KeyNotAvailableError : public std::exception {
     virtual const char* what() const noexcept override { return what_str.c_str(); }
 };
 
+template <typename T>
+bool isNan(const T& field){
```

Comment:
I think this would fit better in FieldTypes.h. Can you move it there and add a comment that we use it to prevent NaNs in fields?